common/spdat.cpp

```diff
@@ -2430,3 +2754,778 @@ bool AegolismStackingIsArmorClassSpell(uint16 spell_id) {
+		}
+		else if (
+			IsBuffSpell(spell_id) &&
+			IsBeneficialSpell(spell_id) &&
+			IsBardSong(spell_id)
+		) {
+			if (
+				spell_type == BotSpellTypes::InCombatBuffSong ||
+				spell_type == BotSpellTypes::OutOfCombatBuffSong ||
+				spell_type == BotSpellTypes::PreCombatBuffSong
+				) {
+				correct_type = spell_type;
+			}
+			else {
+				correct_type = BotSpellTypes::OutOfCombatBuffSong;
+			}
+		}
+		else if (
+			!IsBardSong(spell_id) &&
+			(
+				(IsSelfConversionSpell(spell_id) && spell.buff_duration < 1) ||
+				(spell_type == BotSpellTypes::InCombatBuff && IsAnyBuffSpell(spell_id))
+			)
+		) {
+			correct_type = BotSpellTypes::InCombatBuff;
+		}
+		else if (
+			spell_type == BotSpellTypes::PreCombatBuff &&
+			IsAnyBuffSpell(spell_id) &&
+			!IsBardSong(spell_id)
+		) {
+			correct_type = BotSpellTypes::PreCombatBuff;
+		}
+		else if (
+			(IsCureSpell(spell_id) && spell_type == BotSpellTypes::Cure) ||
+			(IsCureSpell(spell_id) && !IsAnyHealSpell(spell_id))
+		) {
+			correct_type = BotSpellTypes::Cure;
+		}
+		else if (IsAnyNukeOrStunSpell(spell_id)) {
+			if (IsAnyAESpell(spell_id)) {
+				if (IsAERainSpell(spell_id)) {
+					correct_type = BotSpellTypes::AERains;
+				}
+				else if (IsPBAENukeSpell(spell_id)) {
+					correct_type = BotSpellTypes::PBAENuke;
+				}
+				else if (IsStunSpell(spell_id)) {
+					correct_type = BotSpellTypes::AEStun;
+				}
+				else {
+					correct_type = BotSpellTypes::AENukes;
+				}
+			}
+			else if (IsStunSpell(spell_id)) {
+				correct_type = BotSpellTypes::Stun;
+			}
+			else {
+				correct_type = BotSpellTypes::Nuke;
+			}
+		}
+		else if (IsAnyHealSpell(spell_id)) {
+			if (IsGroupSpell(spell_id)) {
+				if (IsGroupCompleteHealSpell(spell_id)) {
+					correct_type = BotSpellTypes::GroupCompleteHeals;
+				}
+				else if (IsGroupHealOverTimeSpell(spell_id)) {
+					correct_type = BotSpellTypes::GroupHoTHeals;
+				}
+				else if (IsRegularGroupHealSpell(spell_id)) {
+					correct_type = BotSpellTypes::GroupHeals;
+				}
+			}
+			else {
+				if (IsVeryFastHealSpell(spell_id)) {
+					correct_type = BotSpellTypes::VeryFastHeals;
+				}
+				else if (IsFastHealSpell(spell_id)) {
+					correct_type = BotSpellTypes::FastHeals;
+				}
+				else if (IsCompleteHealSpell(spell_id)) {
+					correct_type = BotSpellTypes::CompleteHeal;
+				}
+				else if (IsHealOverTimeSpell(spell_id)) {
+					correct_type = BotSpellTypes::HoTHeals;
+				}
+				else if (IsRegularSingleTargetHealSpell(spell_id)) {
+					correct_type = BotSpellTypes::RegularHeal;
+				}
+				else if (IsRegularPetHealSpell(spell_id)) {
+					correct_type = BotSpellTypes::RegularHeal;
+				}
+			}
+		}
+		else if (IsAnyBuffSpell(spell_id)) {
+			if (IsResistanceOnlySpell(spell_id)) {
+				correct_type = BotSpellTypes::ResistBuffs;
+			}
+			else if (IsDamageShieldOnlySpell(spell_id)) {
+				correct_type = BotSpellTypes::DamageShields;
+			}
+			else {
+				correct_type = BotSpellTypes::Buff;
```

Comment:
Instead of using `else {` you can reducing branching by just making it the default at the top

```cpp
correct_type = BotSpellTypes::Buff;
if (IsResistanceOnlySpell(spell_id)) {
	correct_type = BotSpellTypes::ResistBuffs;
}
else if (IsDamageShieldOnlySpell(spell_id)) {
	correct_type = BotSpellTypes::DamageShields;
}
```